### PR TITLE
statistics: fix refactor

### PR DESCRIPTION
**Pull request by Luffbee**
### What problem does this PR solve? 

The previous refactor PR #1750 accidentally removed the change of calculating region flow in PR #1688.

### What is changed and how it works?

Fix it and made some upgrade:
PR #1688 only use the report interval to improve the flow rates.
This PR also use report interval to denoise.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: server/statistics/hot_peer_cache.go [hunk 1]
```diff
@@ -85,24 +85,24 @@ func (f *hotPeerCache) CheckRegionFlow(region *core.RegionInfo, stats *StoresSta
...
 		oldItem := f.getOldHotPeerStat(region.GetID(), storeID)
 
 		// This is used for the simulator.
-		if oldItem != nil && Denoising {
-			interval := time.Since(oldItem.LastUpdateTime).Seconds()
-			// ignore if report too fast
-			if interval < hotRegionReportMinInterval && !isExpired {
+		if oldItem != nil && !isExpired {
+			// ignore if report too fast or an old report
+			isOldReport := endTime.After(oldItem.LastUpdateTime)
```
Comment (rleungx) [inline server/statistics/hot_peer_cache.go]: I'm a little bit confused about the meaning of `isOldReport`, does it mean the report time is before the `LastUpdateTime`?

Comment (Luffbee) [reply server/statistics/hot_peer_cache.go]: Yes, my mistake. Fixed.

Comment (lhy1024) [reply server/statistics/hot_peer_cache.go]: BTW, do we need to add a test to avoid similar problem

**Pull request merged**